tests/trainer/test_dataloaders.py

```diff
@@ -551,10 +538,6 @@ def test_dataloaders_with_fast_dev_run(tmpdir, fast_dev_run):
         trainer.test(model)
         assert trainer.num_test_batches == [fast_dev_run] * len(trainer.test_dataloaders)
 
-        # verify sanity check batches match as expected
-        num_val_dataloaders = len(model.val_dataloader())
-        assert trainer.dev_debugger.num_seen_sanity_check_batches == trainer.num_sanity_val_steps * num_val_dataloaders
-
```

Comment:
reviewer: already covered by the test in `tests/trainer/test_trainer.py::test_num_sanity_val_steps`